components/message.php

```diff
@@ -0,0 +1,504 @@
+<?php
+/**
+ * Manage BuddyPress Messages.
+ *
+ * @since 1.6.0
+ */
+class BPCLI_Message extends BPCLI_Component {
+
+	/**
+	 * Object fields.
+	 *
+	 * @var array
+	 */
+	protected $obj_fields = array(
+		'id',
+		'subject',
+		'message',
+		'thread_id',
+		'sender_id',
+		'date_sent',
+	);
+
+	/**
+	 * Add a message.
+	 *
+	 * ## OPTIONS
+	 *
+	 * --from=<user>
+	 * : Identifier for the user. Accepts either a user_login or a numeric ID.
+	 *
+	 * [--to=<user>]
+	 * : Identifier for the recipient. Accepts either a user_login or a numeric ID.
+	 * ---
+	 * Default: Empty.
+	 * ---
+	 *
+	 * [--subject=<subject>]
+	 * : Subject of the message.
+	 * ---
+	 * Default: Message Subject.
+	 * ---
+	 *
+	 * [--content=<content>]
+	 * : Content of the message.
+	 * ---
+	 * Default: Random content.
+	 * ---
+	 *
+	 * [--thread-id=<thread-id>]
+	 * : Thread ID.
+	 * ---
+	 * Default: false
+	 * ---
+	 *
+	 * [--date-sent=<date-sent>]
+	 * : MySQL-formatted date.
+	 * ---
+	 * Default: current date.
+	 * ---
+	 *
+	 * [--silent=<silent>]
+	 * : Whether to silent the message creation.
+	 * ---
+	 * Default: false.
+	 * ---
+	 *
+	 * [--porcelain]
+	 * : Return the thread id of the message.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp message create --from=user1 --to=user2 --subject="Message Title" --content="We are ready"
+	 *     Success: Message successfully created.
+	 *
+	 *     $ wp bp message create --from=545 --to=313
+	 *     Success: Message successfully created.
+	 *
+	 * @alias add
+	 */
+	public function create( $args, $assoc_args ) {
+		$r = wp_parse_args( $assoc_args, array(
+			'to'        => '',
+			'subject'   => '',
+			'content'   => '',
+			'thread-id' => false,
+			'date-sent' => bp_core_current_time(),
+			'silent'    => false,
+		) );
+
+		// Fill in the content.
+		if ( empty( $r['content'] ) ) {
+			$r['content'] = $this->generate_random_text();
+		}
+
+		// Fill in the subject.
+		if ( empty( $r['subject'] ) ) {
+			$r['subject'] = sprintf( 'Message Subject' );
+		}
+
+		$user = $this->get_user_id_from_identifier( $assoc_args['from'] );
+		if ( ! $user ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		// To is not required when thread id is set.
```

Comment:
This confused me a bit when I first tested - maybe worth mentioning in the function docs, so it shows up in `help`.